tests/unit/machine/FeatureDispatchCRTP_unittest.cc

```diff
@@ -0,0 +1,135 @@
+#include "utils/Utils.h"
+#include <gtest/gtest.h>
+#include <shogun/base/some.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/features/StringFeatures.h>
+#include <shogun/machine/FeatureDispatchCRTP.h>
+#include <shogun/machine/LinearMachine.h>
+
+using namespace shogun;
+
+class CDenseRealMockMachine
+    : public CDenseRealDispatch<CDenseRealMockMachine, CMachine>
+{
+public:
```

Comment:
making it protected caused errors with some<> also train_machine_templated need to be public so i will update it when i look into making it friend